### PR TITLE
CEC-1295 Salesforce Connection Management

### PR DESCRIPTION
**Pull request by Shubhangi-cs**

### CODE REVIEW

File: src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/SalesforceSinkConfig.java [hunk 1]
```diff
@@ -170,7 +190,7 @@ public ErrorHandling getErrorHandling() {
   }
 
   public void validate(Schema schema, FailureCollector collector) {
-    super.validate(collector);
+    getConnection().validate(collector);
```
Comment (vikasrathee-cs) [inline src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/SalesforceSinkConfig.java]: Add null check for get connection at all places.

File: src/main/java/io/cdap/plugin/salesforce/plugin/source/streaming/SalesforceStreamingSourceConfig.java [hunk 2]
```diff
@@ -26,21 +26,25 @@
...
-import io.cdap.plugin.salesforce.plugin.BaseSalesforceConfig;
 import io.cdap.plugin.salesforce.plugin.OAuthInfo;
+import io.cdap.plugin.salesforce.plugin.SalesforceConnectorConfig;
 import io.cdap.plugin.salesforce.soap.SObjectBuilder;
 import io.cdap.plugin.salesforce.soap.SObjectUtil;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.io.Serializable;
+import java.time.temporal.ChronoUnit;
```
Comment (vikasrathee-cs) [inline src/main/java/io/cdap/plugin/salesforce/plugin/source/streaming/SalesforceStreamingSourceConfig.java]: are you using this class?

File: src/main/java/io/cdap/plugin/salesforce/connector/SObjectList.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.plugin.salesforce.connector;
+
+/**
+ * Set and Reads the response of the SObject.
+ */
+public class SObjectList {
```
Comment (sgarg-CS) [inline src/main/java/io/cdap/plugin/salesforce/connector/SObjectList.java]: Where is this class `SObjectList` used?

File: src/main/java/io/cdap/plugin/salesforce/plugin/source/streaming/SalesforceStreamingSourceConfig.java [hunk 4]
```diff
@@ -26,21 +26,25 @@
...
 import java.util.Collections;
+import java.util.Map;
```
Comment (sgarg-CS) [inline src/main/java/io/cdap/plugin/salesforce/plugin/source/streaming/SalesforceStreamingSourceConfig.java]: Remove unused imports.

File: src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/SalesforceSinkConfig.java (hunk 1 above)
Comment (Shubhangi-cs) [reply src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/SalesforceSinkConfig.java]: Added, according to the requirements.

File: src/main/java/io/cdap/plugin/salesforce/plugin/source/streaming/SalesforceStreamingSourceConfig.java (hunk 2 above)
Comment (Shubhangi-cs) [reply src/main/java/io/cdap/plugin/salesforce/plugin/source/streaming/SalesforceStreamingSourceConfig.java]: Removed the import.

File: src/main/java/io/cdap/plugin/salesforce/connector/SObjectList.java (hunk 3 above)
Comment (Shubhangi-cs) [reply src/main/java/io/cdap/plugin/salesforce/connector/SObjectList.java]: Deleted the class as it was unused.

File: src/main/java/io/cdap/plugin/salesforce/plugin/source/streaming/SalesforceStreamingSourceConfig.java (hunk 4 above)
Comment (Shubhangi-cs) [reply src/main/java/io/cdap/plugin/salesforce/plugin/source/streaming/SalesforceStreamingSourceConfig.java]: Removed

**Pull request closed without merging**